Install JSROOT from the build tree when builtin_jsroot is used

With builtin_jsroot=ON, JSROOT (and mathjax with builtin_mathjax=ON) are
unpacked into the js directory of the build tree, but the install rule for
the js directory refers to the source tree, which only contains the few
files in js/files. An installed ROOT therefore had no JSROOT and fell back
to loading it from root.cern.

Install the js directory of the build tree instead, where the files from
the source tree are copied to as well.

---
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 54eccd9ab1b..a06a7adc416 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -665,7 +665,13 @@ if(NOT CMAKE_SOURCE_DIR STREQUAL CMAKE_INSTALL_PREFIX)
   install(DIRECTORY icons/  DESTINATION ${CMAKE_INSTALL_ICONDIR} ${DIR_PERMISSIONS})
   install(DIRECTORY macros/ DESTINATION ${CMAKE_INSTALL_MACRODIR} ${DIR_PERMISSIONS})
   if(http)
-     install(DIRECTORY js/     DESTINATION ${CMAKE_INSTALL_JSROOTDIR} ${DIR_PERMISSIONS})
+    if(builtin_jsroot)
+      # JSROOT (and mathjax) are unpacked into the js directory of the build tree,
+      # where the js directory of the source tree is copied to as well
+      install(DIRECTORY ${CMAKE_BINARY_DIR}/js/ DESTINATION ${CMAKE_INSTALL_JSROOTDIR} ${DIR_PERMISSIONS})
+    else()
+      install(DIRECTORY js/ DESTINATION ${CMAKE_INSTALL_JSROOTDIR} ${DIR_PERMISSIONS})
+    endif()
   endif()
   if(cefweb AND TARGET ROOTCefDisplay)
      install(DIRECTORY ${CMAKE_BINARY_DIR}/Frameworks/   DESTINATION ${CMAKE_INSTALL_FRAMEWORKSDIR} ${DIR_PERMISSIONS})
